### ISSUE TITLE
Support Boolean Variants

### ISSUE THREAD

**Issue by joe-bell** [joe-bell/cva#2]
**Is your feature request related to a problem? Please describe.**

Currently, `cva` doesn't support the use of `boolean` variants unless wrapped in a string:

```ts
const button = cva("button", {
  variants: {
    disabled: {
      false: "opacity-100"
      true: "opacity-050"
    }
  }
});

button({ disabled: "true" })
```

**Describe the solution you'd like**
Allow for boolean variants without string wrapping:

```ts
const button = cva("button", {
  variants: {
    disabled: {
      false: "opacity-100"
      true: "opacity-050"
    }
  }
});

button({ disabled: true })
```

**Additional context**
This is very much my first deep dive into Generics, and I considered this more of a "nice-to-have" rather than something necessary for MVP. I'd love to include it, but it's a bit of a challenge. Feel free to pick this up if you have a better idea of what needs doing 🙏

**Comment by mariusflorescu** [comment]
Will there be any cases where we want to have the ability to have both `boolean` and `string` keys in the same variant?
Such as:

```js
const button = cva("button", {
  variants: {
    random: {
      false: "opacity-100",
      semi: "opacity-050",
      true: "opacity-0"
    }
  }
});

button({ disabled: true });
button({ disabled: "semi"});
```

**Comment by joe-bell** [comment]
@mariusflorescu I think ideally I'd like to limit to just `boolean` or `string – but that's probably one for a major release?

**Comment by joe-bell** [comment]
Closed by #8
